src/proof_of_possession.rs

```diff
@@ -195,9 +203,11 @@ impl ProofOfPossession {
         let (controller, jwk) = match (header.key_id, header.jwk, header.x509_certificate_chain) {
             (Some(kid), None, None) => {
                 let vm = kid.parse()?;
-                get_jwk_from_kid(&kid, resolver)
+                //get_jwk_from_kid(&kid, resolver)
+                resolver
+                    .fetch_public_jwk(Some(&kid))
                     .await
-                    .map(|r| (Some(vm), r))?
+                    .map(|r| (Some(vm), r.deref().clone()))?
```

Comment:
`r` is a `Cow<JWK>`, you should use `r.into_owned()` here to avoid cloning when `r = Cow::Owned(jwk)`.